node-hub/opencv-plot/main.py

```diff
@@ -0,0 +1,164 @@
+import os
+import argparse
+import cv2
+
+import numpy as np
+import pyarrow as pa
+
+from dora import Node
+
+
+class Plot:
+    frame: np.array = np.array([])
+
+    bboxes: {} = {
+        "bbox": np.array([]),
+        "conf": np.array([]),
+        "names": np.array([]),
+    }
+
+    text: str = ""
+
+    width: np.uint32 = None
+    height: np.uint32 = None
+
+
+def plot_frame(plot, ci_enabled):
+    for bbox in zip(plot.bboxes["bbox"], plot.bboxes["conf"], plot.bboxes["names"]):
+        [
+            [min_x, min_y, max_x, max_y],
+            confidence,
+            label,
+        ] = bbox
+        cv2.rectangle(
+            plot.frame,
+            (int(min_x), int(min_y)),
+            (int(max_x), int(max_y)),
+            (0, 255, 0),
+            2,
+        )
+
+        cv2.putText(
+            plot.frame,
+            f"{label}, {confidence:0.2f}",
+            (int(max_x) - 120, int(max_y) - 10),
+            cv2.FONT_HERSHEY_SIMPLEX,
+            0.5,
+            (0, 255, 0),
+            1,
+            1,
+        )
+
+    cv2.putText(
+        plot.frame,
+        plot.text,
+        (20, 20),
+        cv2.FONT_HERSHEY_SIMPLEX,
+        0.5,
+        (255, 255, 255),
+        1,
+        1,
+    )
+
+    if plot.width is not None and plot.height is not None:
+        plot.frame = cv2.resize(plot.frame, (plot.width, plot.height))
+
+    if not ci_enabled:
+        if len(plot.frame.shape) >= 3:
+            cv2.imshow("Dora Node: opencv-plot", plot.frame)
+
+            if cv2.waitKey(1) & 0xFF == ord('q'):
+                return True
+
+    return False
+
+
+def main():
+    # Handle dynamic nodes, ask for the name of the node in the dataflow, and the same values as the ENV variables.
+    parser = argparse.ArgumentParser(
+        description="OpenCV Plotter: This node is used to plot text and bounding boxes on an image.")
+
+    parser.add_argument("--name", type=str, required=False, help="The name of the node in the dataflow.",
+                        default="opencv-plot")
+    parser.add_argument("--plot-width", type=int, required=False, help="The width of the plot.", default=None)
+    parser.add_argument("--plot-height", type=int, required=False, help="The height of the plot.", default=None)
+
+    args = parser.parse_args()
+
+    plot_width = os.getenv("PLOT_WIDTH", args.plot_width)
+    plot_height = os.getenv("PLOT_HEIGHT", args.plot_height)
+
+    if plot_width is not None:
+        if isinstance(plot_width, str) and plot_width.isnumeric():
+            plot_width = int(plot_width)
+
+    if plot_height is not None:
+        if isinstance(plot_height, str) and plot_height.isnumeric():
+            plot_height = int(plot_height)
+
+    # check if the code is running in a CI environment (e.g. GitHub Actions) (parse to bool)
+    ci_enabled = os.getenv("CI", False)
```

Comment:
```suggestion
    RUNNER_CI = os.getenv("CI", "false")
```

Can we rename this `RUNNER_CI` as `ci_enabled` seems like it is a feature flag. Also, os.getenv seems to want a string default

Reply:
Allright!

Reply:
```Python
>>> import os
>>> os.getenv("TEST", 4)
4
>>> type(os.getenv("TEST", 4))
<class 'int'>
``` 
i'm changing for RUNNER_CI but it seems i can keep the boolean value